module/zfs/spa_errlog.c

```diff
@@ -936,6 +936,17 @@ spa_upgrade_errlog(spa_t *spa, dmu_tx_t *tx)
 		sync_upgrade_errlog(spa, spa->spa_errlog_scrub, &newobj, tx);
 		spa->spa_errlog_scrub = newobj;
 	}
+
+	/*
+	 * Update the MOS to reflect the new values.
+	 */
+	(void) zap_update(spa->spa_meta_objset, DMU_POOL_DIRECTORY_OBJECT,
+	    DMU_POOL_ERRLOG_LAST, sizeof (uint64_t), 1,
+	    &spa->spa_errlog_last, tx);
+	(void) zap_update(spa->spa_meta_objset, DMU_POOL_DIRECTORY_OBJECT,
+	    DMU_POOL_ERRLOG_SCRUB, sizeof (uint64_t), 1,
+	    &spa->spa_errlog_scrub, tx);
+
```

Comment:
Nit: you could optimize this slightly by only updating the zap fields in the `DMU_POOL_DIRECTORY_OBJECT` if you actually call `sync_upgrade_errlog()` for that field.

Reply:
Yes, I see!